### PR TITLE
MCKIN-12757 fix courses count by excluding company admins

### PR DESCRIPTION
**Pull request by mudassir-hafeez**
Fixed organization courses count

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @mudassir-hafeez! I've created [OSPR-4277](https://openedx.atlassian.net/browse/OSPR-4277) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).



We can't start reviewing your pull request until you've submitted a     [signed contributor agreement](https://open.edx.org/wp-content/uploads/2019/01/individual-contributor-agreement.pdf)     or indicated your institutional affiliation. Please see the [CONTRIBUTING](https://github.com/edx/edx-platform/blob/master/CONTRIBUTING.rst)     file for more information.

File: edx_solutions_organizations/views.py [hunk 1]
```diff
@@ -66,13 +68,43 @@ def list(self, request, *args, **kwargs):
...
+            q_object.add(~Q(users__courseenrollment__user_id__in=user_ids), Q.AND)
+
+            admin_users = list(queryset.filter(
+                Q(users__groups__groupprofile__name=exclude_type)
+            ).values_list('id', 'users'))
+
+            admin_users_dic = {}
+            for id, user_id in admin_users:
+                user = User.objects.get(id=user_id)
+                main_company = user.organizations.all()[0].id
```
Comment (nasirhjafri) [inline edx_solutions_organizations/views.py]: Get all users in a single query and use prefetch for organizations

File: edx_solutions_organizations/views.py [hunk 2]
```diff
@@ -66,13 +68,43 @@ def list(self, request, *args, **kwargs):
...
+            roles_to_exclude = [CourseInstructorRole.ROLE, CourseStaffRole.ROLE, CourseObserverRole.ROLE, CourseAssistantRole.ROLE]
+            user_ids = CourseAccessRole.objects.filter(role__in=roles_to_exclude).distinct().values_list('user_id', flat=True)
+            q_object.add(~Q(users__courseenrollment__user_id__in=user_ids), Q.AND)
+
+            admin_users = list(queryset.filter(
+                Q(users__groups__groupprofile__name=exclude_type)
+            ).values_list('id', 'users'))
+
+            admin_users_dic = {}
+            for id, user_id in admin_users:
```
Comment (nasirhjafri) [inline edx_solutions_organizations/views.py]: id is a builtin.

**Review by nasirhjafri — CHANGES_REQUESTED** [review]
Please add comments on the code. Thanks @mudassir-hafeez

File: edx_solutions_organizations/views.py (hunk 1 above)
Comment (mudassir-hafeez) [reply edx_solutions_organizations/views.py]: @nasirhjafri thanks for reviewing. Actually, prefetch changes the order of the organizations and we need the order here. so I refactored the code a bit and added few comments.

File: edx_solutions_organizations/views.py [hunk 3]
```diff
@@ -66,13 +68,44 @@ def list(self, request, *args, **kwargs):
...
+
+            admin_users_dict = {}
+            for user in admin_users:
+                other_companies = list(user.organizations.all())[1:]
+                for company in other_companies:
+                    admin_users_dict.setdefault(company.id, []).append(user.id)
+
+            exclude_admin_users = [
+                Q(id=k) & Q(users__courseenrollment__user_id__in=v) for
+                k, v in admin_users_dict.items()]
```
Comment (musmanmalik) [inline edx_solutions_organizations/views.py]: Please use meaningful variables.

Comment (mudassir-hafeez) [reply edx_solutions_organizations/views.py]: Addressed

**Review by nasirhjafri — APPROVED** [review]

**Pull request merged**